docs/build-process/worker-build-process.md

```diff
@@ -0,0 +1,90 @@
+
+We provide a reference implementation for building a C# worker that includes all worker build steps described in this guide. You can find it in the form of a shell script [here](../../build.sh).
+
+## Worker build process overview
+To build a worker in the FPL, you need to:
+1. [Download dependencies](#1-download-dependencies)
+2. [Generate code](#2-generate-code)
+3. [Build the worker executable](#3-build-the-worker-executable)
+
+**Migration advice:** With the CPL, the spatial CLI performs the worker build steps listed above through a combination of CLI commands ([`spatial worker codegen`](https://docs.improbable.io/reference/latest/shared/spatial-cli/spatial-worker-codegen), [`spatial worker build`](https://docs.improbable.io/reference/latest/shared/spatial-cli/spatial-worker-build)) and configuration files ([`worker build configuration`](https://docs.improbable.io/reference/latest/shared/worker-configuration/worker-build), [`spatialos_worker_packages.json`](https://docs.improbable.io/reference/latest/shared/reference/file-formats/spatial-worker-packages)). With the FPL, you need to perform all build steps manually which gives you full control over the build process.
+
+## 1. Download dependencies
+A SpatialOS SDK release consists of a collection of artifacts (libraries, tools etc.). In this build step, you will download the necessary artifacts for building a worker.
+
+**Tip:** You may want to add the dependencies mentioned in this build step to your `.gitignore` file or equivalent to avoid checking them into source control.
+
+### Choose an SDK version
+Before you can download any dependencies, you need to be mindful about the SDK version that you want to build your worker with. 
+* If you are upgrading a worker from the CPL to the FPL, we recommend you use the SDK version that your project is currently using as specified in your [`project definition file (CPL format)`](https://docs.improbable.io/reference/latest/shared/reference/file-formats/spatialos-json) to reduce the complexity of the migration process.
+* If you are creating a worker from scratch, we recommend that you use the same SDK version as other workers in your project.
+
+**Notes:** 
+* You can mix SDK versions in an FPL project - in other words, you can build different workers within the same project using different SDK versions. This can be useful when transitioning a project to a newer SDK version - for example, you may opt to upgrade the SDK version of one worker at a time instead of upgrading the full project all at once.
+* Since there is no clearly defined SDK version for an FPL project, we recommend that you manually keep track of the SDK version(s) you are using to avoid forgetting that information.
```

Comment:
Hm, so there's no way to figure out the SDK version used for a given worker if an user really forgets about it? Seems like a future feature ticket :)

Reply:
Yes, this could potentially lead to problems down the road. The problem here is that since you have full control over the worker build process, we cannot control what dlls you use to build your workers with (it could be any sdk version and there is no way for us to tell). @improbablepiotr has plans to let workers report the sdk version they are using to fabric as they connect to a deployment as a means to determine the sdk version of your project. We might maybe also consider adding a util method in the sdk dlls to retrieve the version without connecting (dll could provide a `GetVersion()` method).